lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -918,15 +939,49 @@ def rotate_axes(xs, ys, zs, zdir):
         return xs, ys, zs
 
 
-def _zalpha(colors, zs):
+def _zalpha(colors, zs, _dscl=None):
     """Modify the alphas of the color list according to depth."""
-    # FIXME: This only works well if the points for *zs* are well-spaced
-    #        in all three dimensions. Otherwise, at certain orientations,
-    #        the min and max zs are very close together.
-    #        Should really normalize against the viewing depth.
+    # Return empty if given empty
     if len(colors) == 0 or len(zs) == 0:
         return np.zeros((0, 4))
-    norm = Normalize(min(zs), max(zs))
-    sats = 1 - norm(zs) * 0.7
+
+    if _dscl is None:
+        # This only works well if the points for *zs* are well-spaced
+        # in all three dimensions. Otherwise, at certain orientations,
+        # the min and max zs are very close together.
+        # Should really normalize against the viewing depth.
+
+        # Normalize the z-depths to the range 0 - 1
+        norm = Normalize(min(zs), max(zs))
+
+        # Generate alpha multipliers using the normalized z-depths so that
+        # closer points
+        # are opaque and the furthest points are still visible, but transparent
+        sats = 1 - norm(zs) * 0.7
+
+    else:
+        # Improved normalization using a scale value derived from the XYZ
+        # limits of the plot
+
+        # Solid near, transparent far, solid default
+        if _dscl == 0:
+            sats = np.ones_like(zs)
+        else:
+            sats = np.clip(1 - (zs - min(zs)) / _dscl, 0.3, 1)
+        # print(_dscl)
+
+        # Solid near, transparent far, transparent default
+        # sats = np.clip((max(zs)-zs)/(_dscl+1e-8), 0.3, 1)
+
+        # Transparent near, solid far, solid default
+        # sats = np.clip(1-(max(zs)-zs)/(_dscl+1e-8), 0.3, 1)
+
+        # Transparent near, solid far, transparent default
+        # sats = np.clip((zs-min(zs))/(_dscl+1e-8), 0.3, 1)
+
+    # Restructure colors into a rgba numpy array
     rgba = np.broadcast_to(mcolors.to_rgba_array(colors), (len(zs), 4))
+
+    # Change the alpha values of the colors using the generated alpha
+    # multipliers
```

Comment:
Also somewhat redundant.

Reply:
I would prefer to leave it in just to be explicit. With the indexing and column stacking the intent might not be obvious at a glance